### ISSUE TITLE
[BUG] [API Server VNet Integration] "private DNS zone" is still preserved after disabling private cluster mode

### ISSUE THREAD

**Issue by JoeyC-Dev** [Azure/AKS#4297]
**Describe the bug**
Private DNS zone is still preserved after disabling private cluster mode while using "API Server VNet Integration".

**To Reproduce**
```bash
aks=testAKS
rG=resourceGroup
location=eastus

az group create -n ${rG} -l ${location}

# Initial creation of public cluster with API Server VNet Integration
az aks create --name ${aks} --resource-group ${rG} --network-plugin azure --enable-apiserver-vnet-integration --no-ssh-key

# Enable private cluster mode
az aks update --name ${aks} --resource-group ${rG} --enable-private-cluster

# Disable private cluster mode
az aks update --name ${aks} --resource-group ${rG} --disable-private-cluster

# Clean resources
az group delete -n ${rG} --no-wait
```

**Expected behavior**

Private DNS Zone should be deleted as private cluster mode is disabled. 

**Screenshots**
1. Initial creation of public cluster with API Server VNet Integration
![image](https://github.com/Azure/AKS/assets/142381267/fbe67d77-c932-430f-a630-b4327ef1cf63)

2. Enable private cluster mode
![image](https://github.com/Azure/AKS/assets/142381267/05e05cf5-c3fb-4b30-b33a-c96037591632)

3. Disable private cluster mode
![image](https://github.com/Azure/AKS/assets/142381267/abcb70b8-c1f5-4316-a8ce-eb24f837467a)

**Environment (please complete the following information):**
 -  CLI Version [e.g. 3.22]: 2.60.0
 - Kubernetes version [e.g. 1.24.3]: 1.28.9
 - CLI Extension version [e.g. 1.7.5] if applicable: aks-preview 3.0.0b13
 - Browser [e.g. chrome, safari] is applicable: N/A

**Additional context**
None

**Comment by levimm** [comment]
would fix this in AKS September release.

**Comment by levimm** [comment]
fixed with AKS 20240918 release, closing this one.
